### PR TITLE
sites compatibility

### PR DESCRIPTION
**Pull request by reepush**
moved tests which fail in CI but pass manually to CIFails suite, so we can better distinguish which tests need attention

### CODE REVIEW

File: sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature [hunk 1]
```diff
@@ -16,4 +16,39 @@ When I login Indiegogo with 'citesting@themooltipass.com'
...
+Scenario: Testing ldlc.ch
+Given I navigate to 'http://www.ldlc.ch/'
+When I login IDLC with 'citesting@themooltipass.com'
+Then I should be logged in IDLC
+When I logout IDLC
+And I go to IDLC login page
+Then I should be logged in IDLC
+
+@supported
+Scenario: Testing Techmania
```
Comment (limpkin) [inline sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature]: from @mosaadm : "techmania and tripadvisor are still failing" >> so why should it go there? it's still not supported.

File: sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature [hunk 2]
```diff
@@ -16,4 +16,39 @@ When I login Indiegogo with 'citesting@themooltipass.com'
...
+Given I navigate to 'https://techmania.ch/'
+When I log in Techmania with 'mooltipass'
+Then I should be logged in Techmania
+When I logout Techmania
+When I go to Techmania login page
+Then I should be logged in Techmania
+
+@supported
+Scenario: Testing tripadvisor.com
```
Comment (limpkin) [inline sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature]: from @mosaadm : "techmania and tripadvisor are still failing" >> so why should it go there? it's still not supported.

File: sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature [hunk 3]
```diff
@@ -16,4 +16,39 @@ When I login Indiegogo with 'citesting@themooltipass.com'
...
+Given I navigate to 'https://www.tripadvisor.com'
+When I login tripadvisor with 'citesting@themooltipass.com'
+Then I should be logged in tripadvisor
+When I logout tripadvisor
+And I go to tripadvisor login page
+Then I should be logged in tripadvisor
+
+@supported
+Scenario: Testing pc-ostschweiz.ch
```
Comment (limpkin) [inline sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/CIFails.feature]: why is it there?

File: sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/Supported.feature [hunk 4]
```diff
@@ -214,3 +214,39 @@ When I logout ProtonMail
...
+Scenario: Testing https://workbench.cisecurity.org/
+Given I navigate to 'https://workbench.cisecurity.org/'
+When I login cisecurity with 'citesting@themooltipass.com'
+Then I should be logged in cisecurity
+When I logout cisecurity
+And I go to cisecurity login page
+Then I should be logged in cisecurity
+
+@supported
+Scenario: Testing https://www.trillian.im/web/4.0/
```
Comment (limpkin) [inline sauce_labs_tests/src/test/resources/mooltipass/automatedTest/features/Supported.feature]: no! when manually testing it we get constant requests

**Comment by limpkin** [comment]
@reepush : it seems you didn't read @mosaadm who mentioned that there's nothing to put in CI fails! we need to talk as soon as possible.

**Comment by reepush** [comment]
@mosaadm 
I checked lcdc.ch again, I have adblock installed that's why that link button didn't show up for me. Thanks for the feedback. Now it's fixed. I can move this test to supported after you test and approve it. Will it work this way?

**Comment by reepush** [comment]
@mosaadm  
And with techmania.ch I get a strange behaviour – it works manually, but in automated test _onSubmit_ handler isn't triggered. It can be caused by our "syntetic events", especially because it's `<a>` element and it's behave a bit different as I discovered today. I will try to submit form by simulating pressing return key on password field, maybe it will work out.

**Comment by limpkin** [comment]
@reepush didn't we agree that we actually can't simulate return key press in the password field?

**Comment by limpkin** [comment]
@reepush techmania.ch does NOT work manually

**Comment by reepush** [comment]
Walked through all tests in NotSupported again:

**ldlc.ch** (problem with ad)
* ✅ Manual
* ✅ CI

**techmania.ch** (problem with link as anchor)
* ✅ Manual
* ❌ CI (click doesn't trigger submit)

**evernote.com** (problem with two-step auth)
* ✅ Manual
* ❌ CI (problem with logout)

**tripadvisor.com** (problem with onSubmit event)
* ✅ Manual
* ✅ CI

**pc-ostschweiz.ch** (problem with onSubmit event)
* ✅ Manual
* ✅ CI

**kenwoodworld.com** (problem with submit button detection)
* ✅ Manual
* ✅ CI

**workbench.cisecurity.org** (problem with submit button detection)
* ✅ Manual
* ✅ CI

**trillian.im (problem with constant requests as @limpkin said, not fixed)
* ✅ Manual
* ✅ CI

**minfin.com.ua** (problem with captcha detection)
* ✅ Manual (form isn't submitted because of wrong detected captcha, maybe no a big deal)
* ❌ CI (no first login)

**microchipdirect.com** (problem with submit button detection)
* ✅ Manual
* ✅ CI

**airbnb.com** (problem with text input)
* ❌ Manual (react specific way of handling inputs, don't even know if we can handle it)
* ❌ CI (chrome is frozen at entering credentials)

**Comment by reepush** [comment]
[Techmania screen recording](https://www.dropbox.com/s/32koxz11awm2izd/techmania-test.mp4?dl=0)

@limpkin 
> this is starting to be a lot of hardcoded website support, which I don't like

Agree. I would like to do two things in this PR:
1. Move from handling submit event to handling click event on submit button and keypress event on password field. This way I can delete all these edge cases.
2. Improve how current submit button detection works – prioritize detected buttons base on how close they are from input fields. It will solve many potential problems in the future.

**Comment by limpkin** [comment]
@reepush 
1. can't we intercept both, have submit event catching as a backup?
2. that makes sense

**Comment by reepush** [comment]
@limpkin 
> can't we intercept both, have submit event catching as a backup?

sure, same way how `doSubmit` currently works with triggering submit event

**Comment by reepush** [comment]
@limpkin 
Haven't figured out problem with airbnb.com and React way of handling input's value. Everything else works and ready for review.

**Comment by limpkin** [comment]
@reepush ok! what's the link to your travis?

**Comment by reepush** [comment]
@limpkin 
[Travis build](https://travis-ci.org/reepush/mooltipass/builds/259898875). Not sure about gmail though.

File: chrome_extension/vendor/mooltipass/mcCombinations.js [hunk 5]
```diff
@@ -893,13 +912,21 @@ mcCombinations.prototype.getAllForms = function() {
...
+	}.bind(this), 100)
+	
 	if (this.settings.debugLevel > 1) cipDebug.log('%c mcCombinations: %c onSubmit','background-color: #c3c6b4','color: #333333');
 	this.waitingForPost = false;
 
 	// Check if there's a difference between what we retrieved and what is being submitted
-	var currentForm = this.forms[ mpJQ(event.target).data('mp-id') ];
-
-	if ( currentForm.combination && !currentForm.combination.savedFields.username && this.credentialsCache) {
+	if ( !currentForm.combination.savedFields.username && this.credentialsCache) {
```
Comment (limpkin) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: currentForm.combination may be empty no? I actually remember writing this line.

**Comment by limpkin** [comment]
might be worth updating the documentation of mccombinations!

File: chrome_extension/vendor/mooltipass/mcCombinations.js (hunk 5 above)
Comment (reepush) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: Yes, I have this check on line 916:
> if (!currentForm.combination) return

Comment (limpkin) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: oh right!

**Comment by reepush** [comment]
@limpkin 
I have updated _mcCombinations.md_

**Comment by limpkin** [comment]
@reepush , from @mosaadm : 

 - techmania-> had a problem and i fixed it so now it works
- tripadvisior-> is working but the login fields are not detected on saucelabs. The test passes when running on local machine
- minfin -> there was change to website and test updated but mooltipass only writes the credentials and does not press login.
- airbnb -> robot was detected. We need to confirm email again
regarding tripadvisior it seems that the login fields are not always detected for some reason, if you look at last few seconds of video here you can see they are not detected. https://saucelabs.com/beta/tests/99dfedb94f5d4e95aa8902c31db78594/watch#38

**Comment by limpkin** [comment]
i confirm once again that for me techmania doesn't work manually.

**Pull request merged**